core/api/service/state/requests/get_runtime_version.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "api/service/state/requests/get_runtime_version.hpp"
+
+namespace kagome::api::state::request {
+
+  GetRuntimeVersion::GetRuntimeVersion(std::shared_ptr<StateApi> api)
+      : api_(std::move(api)) {
+    BOOST_ASSERT(!!api_);
```

Comment:
I would rather avoid such complex constructs where you could explicitly compare it with nullptr.

Reply:
```operator!``` is the common way to check objs validity and it independent of objs nature, whereas ```!= nullptr``` is nature dependent and valid in ptrs context only.

Reply:
Here we know that api is shared_ptr. So comparing with nullptr is not criminal and more common way to assure that api is not empty